lib/blockchain.go

```diff
@@ -88,10 +88,16 @@ func (nn *BlockNode) IsValidated() bool {
 	return nn.Status&StatusBlockValidated != 0
 }
 
+// IsValidateFailed returns true if a BlockNode has failed validations. A BlockNode that is validate failed
+// will never be added to the best chain.
+func (nn *BlockNode) IsValidateFailed() bool {
+	return nn.Status&StatusBlockValidateFailed != 0
+}
+
 // IsCommitted returns true if a BlockNode has passed all validations, and it has been committed to
 // the Blockchain according to the Fast HotStuff commit rule.
 func (nn *BlockNode) IsCommitted() bool {
-	return nn.Status&StatusBlockCommitted != 0
+	return nn.Status&StatusBlockCommitted != 0 || !blockNodeProofOfStakeCutoverMigrationTriggered(nn.Height)
```

Comment:
Why do we need the block height check? If the COMMITTED flag was explicitly set on the node by some caller, then this Getter function should reflect that.

Reply:
This allows us to treat ALL PoW blocks as COMMITTED without updating the PoW Process Block logic and going through all PoW blocks and marking them as committed.

Reply:
Ah, makes sense. This is my mistake, I misread the condition. Please ignore my comment